src/main/python/ayab/engine/engine_fsm.py

```diff
@@ -74,7 +82,7 @@ def _API6_connect(control, operation):
             if operation == Operation.KNIT:
                 control.com = CommunicationMock()
             else:
-                control.com = HardwareTestCommunicationMock()
+                control.com = HardwareTestCommunicationMock() #type: ignore
```

Comment:
As a rule, I didn't dig into the mocks typing for this PR. It's a lot of hassle for little gain.